### ISSUE TITLE
Signature Control Addin Not working In BC

### ISSUE THREAD

**Issue by Akil123** [Microsoft/AL#4220]
Hi All,

We have Developed Signature control Add-in in AL VS code. As below

page 5000 "Sales Header Signature"
{
    Caption = 'Signature';
    PageType = CardPart;
    SourceTable = "Sales Header";

    layout
    {
        area(content)
        {
            group(Signature)
            {
                usercontrol(ControlName; "GAC Signature Addin")
                {
                    ApplicationArea = All;
        
                    
                    trigger AddInReady()
                    begin
                        AddInReady := true;
                        UpdateSignature();
                    end;

                    trigger UpdateSignature()
                    begin
                        UpdateSignature();
                    end;

                    trigger SaveSignature(signatureData: Text)
                    begin
                        Savesignature(signatureData);
                    end;
                }
            }
        }
    }

    trigger OnAfterGetCurrRecord();
    begin
        IF AddInReady THEN
            UpdateSignature();
    end;

    trigger OnAfterGetRecord();
    begin
        IF AddInReady THEN
            UpdateSignature;
    end;

    procedure UpdateSignature();
    begin
        CalcFields("GAC Signature Image");
        if "GAC Signature Image".HasValue() then
            CurrPage.ControlName.PutSignature(GetDataUriFromImage())
        Else
            CurrPage.ControlName.ClearSignature();
    end;

    local procedure SaveSignature(Signaturedata: Text);
    begin
        EncodeValue(signaturedata);
    end;

    local procedure EncodeValue(signaturedata: Text)
    var
        Base64Convert: Codeunit "GAC Signature Base64 Convert";
        OutPutStream: OutStream;
        InputStream: InStream;
        DataImageFromat: Text;
        Txt: Text;
        
    begin
        DataImageFromat := 'data:image/png;base64,';
        Txt := CopyStr(signaturedata, StrLen(DataImageFromat) + 1);
        Rec."GAC Signature Image".CreateOutStream(OutPutStream);
        Base64Convert.ConvertFromBase64StringToStream(Txt, OutPutStream);
        Rec.Modify();
        UpdateSignature();
        CurrPage.Update(false);
    end;

    local procedure GetDataUriFromImage() DataUri: Text;
    var
        Base64Convert: Codeunit "GAC Signature Base64 Convert";
        Inputstream: InStream;
        
    begin
        DataUri := 'data:image/png';
        "GAC Signature Image".CreateInStream(Inputstream);
        DataUri += ';base64,' + Base64Convert.ConvertStreamToBase64String(Inputstream);
    end;

    var
        AddInReady: Boolean;

}

When published & Installed above Control  Addin in Business central i got an below issue.

![image](https://user-images.githubusercontent.com/33834544/48622114-b9d24c00-e9cb-11e8-9980-4871054423d2.png)

But it was worked in NAV 2018 all versions. In NAV 2018 also i have published same app and it works good.

Here is the ZIP file i have used Scripts.

[StyleSheets.zip](https://github.com/Microsoft/AL/files/2589340/StyleSheets.zip)

Please let us help to resolve above problem.

Thank You
Ramesh.

**Comment by StanislawStempin** [comment]
Can you share the declaration for the "GAC Signature Addin"? In Business Central SaaS only pure javascript addins are supported.

**Comment by Akil123** [comment]
Hi,


Please find below zip file for GAC Signature Addin declaration. we have used pure javascript for these development.

[GAC signature.zip](https://github.com/Microsoft/AL/files/2594052/GAC.signature.zip)


let us know  if any thing wrong.

Thank you
Ramesh

**Comment by Akil123** [comment]
Hi Stempin,

Any update on above issue.

Please suggest us.

Thank you
Ramesh
--

**Comment by Akil123** [comment]
Can you please give us source code and also which version it works fine? we tried on BC RTM.

any thing wrong in my Js and css files?

Please check my source code also. 

[ALProject3.zip](https://github.com/Microsoft/AL/files/2694864/ALProject3.zip)

Please suggest us.

Thank you
Ramesh

**Comment by esbenk** [comment]
I have tested this with bcinsider.azurecr.io/bcsandbox-master docker image. I used the original two source snippets in the same projects just deleted everying that didn't compile.

See https://blogs.msdn.microsoft.com/freddyk/2018/04/16/which-docker-image-is-the-right-for-you/ for more details on docker images.

I deleted your previous post with the entire source:-) Maybe a bit to much for this audience?
